crates/client/l1-gas-price/src/worker.rs

```diff
@@ -88,18 +97,49 @@ async fn update_gas_price(
         16,
     )?;
 
-    // TODO: fetch this from the oracle
-    let eth_strk_price = 2425;
+    let response = reqwest::Client::new()
+        .get(oracle.get_fetch_url(String::from("eth"), String::from("strk")))
+        .header("x-api-key", oracle.get_api_key())
+        .send()
+        .await?;
```

Comment:
You don't have a rust SDK yet for Pragma?

Reply:
no we don't